### PR TITLE
fix: step name not accessible to distributor class

### PR DESCRIPTION
**Pull request by alexferrante**
turns out `self.step_name` isn't accessible to `Distributor`. instead of hard-coding the step names I made a function that extracts the step name from `rel_path_to_python_file`. due to order of operations, this requires passing the `rel_path_to_python_file` argument to the constructor. let me know what you think

### CODE REVIEW

File: cvapipe_analysis/tools/cluster.py [hunk 1]
```diff
@@ -35,6 +36,9 @@ def __init__(self, control):
         self.abs_path_to_cvapipe = Path(
             os.path.abspath(cvapipe_analysis.__file__)).parents[1]
 
+    def extract_step_name(self):
+        return self.rel_path_to_python_file.split("/")[-2]
```
Comment (vianamp) [inline cvapipe_analysis/tools/cluster.py:40]: A more elegant way to do this using pathlib: `return Path(self.rel_path_to_python_file).parent.name`. Just keep in mind that both this and your approach would return `compute_features` and not the real step name, which is `computefeatures`. I think it is fine for logging stuff.

Comment (alexferrante) [reply cvapipe_analysis/tools/cluster.py]: I agree with you. I pushed a slight improvement.

**Review by vianamp — COMMENTED** [review]
I am fine with this but also considering another approach which is the main function pass self to distributor so that inside distributor you have access to step_name. What do you think?

**Comment by alexferrante** [comment]
My preference is generally whatever results in the least change to the current code base. It sounds like your idea might involve less change. But this change is not too big either. Up to you

**Comment by vianamp** [comment]
> My preference is generally whatever results in the least change to the current code base. It sounds like your idea might involve less change. But this change is not too big either. Up to you

I think passing self is also more general in case we want to access other attributes of that class in the future.

File: cvapipe_analysis/steps/compute_features/compute_features_tools.py [hunk 2]
```diff
@@ -123,11 +123,11 @@ def get_basic_features(self, img):
                 features[f'roundness_surface_area{suffix}'] = np.nan
         return features
 
-    def get_coeff_features(self, img):
+    def get_coeff_features(self, img, alias):
         (coeffs, _), (_, _, _, transform) = shparam.get_shcoeffs(
             image=img,
             lmax=self.control.get_lmax(),
-            sigma=self.control.get_sigma(),
+            sigma=self.control.get_sigma()[alias],
```
Comment (vianamp) [inline cvapipe_analysis/steps/compute_features/compute_features_tools.py:130]: Shouldn't alias be an argument of `get_sigma`?

**Review by vianamp — APPROVED** [review]
Great!

**Pull request merged**